sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -537,10 +603,137 @@ def peek_messages(self, max_message_count=1, **kwargs):
+        :raises: ~azure.servicebus.exceptions.MessageLockExpired if message lock has already expired.
+        :raises: ~azure.servicebus.exceptions.MessageSettleFailed if message settle operation fails.
+        """
+        if not isinstance(message, ServiceBusReceivedMessage):
+            raise TypeError("Parameter 'message' must be of type ServiceBusReceivedMessage")
+        self._check_message_alive(message, MESSAGE_ABANDON)
+        self._settle_message_with_retry(message, MESSAGE_ABANDON)
+        message._settled = True  # pylint: disable=protected-access
+
+    def defer_message(self, message):
+        """Defers the message.
+
+        This message will remain in the queue but must be requested
+        specifically by its sequence number in order to be received.
+
+        :param message: The received message to be deferred.
+        :type message: ~azure.servicebus.ServiceBusReceivedMessage
+        :rtype: None
+        :raises: ~azure.servicebus.exceptions.MessageAlreadySettled if the message has been settled.
+        :raises: ~azure.servicebus.exceptions.MessageLockExpired if message lock has already expired.
+        :raises: ~azure.servicebus.exceptions.MessageSettleFailed if message settle operation fails.
+        """
+        if not isinstance(message, ServiceBusReceivedMessage):
+            raise TypeError("Parameter 'message' must be of type ServiceBusReceivedMessage")
+        self._check_message_alive(message, MESSAGE_DEFER)
+        self._settle_message_with_retry(message, MESSAGE_DEFER)
+        message._settled = True  # pylint: disable=protected-access
+
+    def dead_letter_message(self, message, reason=None, error_description=None):
+        """Move the message to the Dead Letter queue.
+
+        The Dead Letter queue is a sub-queue that can be
+        used to store messages that failed to process correctly, or otherwise require further inspection
+        or processing. The queue can also be configured to send expired messages to the Dead Letter queue.
+
+        :param message: The received message to be dead-lettered.
+        :type message: ~azure.servicebus.ServiceBusReceivedMessage
+        :param Optional[str] reason: The reason for dead-lettering the message.
+        :param Optional[str] error_description: The detailed error description for dead-lettering the message.
+        :rtype: None
+        :raises: ~azure.servicebus.exceptions.MessageAlreadySettled if the message has been settled.
+        :raises: ~azure.servicebus.exceptions.MessageLockExpired if message lock has already expired.
+        :raises: ~azure.servicebus.exceptions.MessageSettleFailed if message settle operation fails.
+        """
+        if not isinstance(message, ServiceBusReceivedMessage):
+            raise TypeError("Parameter 'message' must be of type ServiceBusReceivedMessage")
+        self._check_message_alive(message, MESSAGE_DEAD_LETTER)
```

Comment:
I'm wondering if the _check_message_alive should go inside _settle_message_with_retry since we have the consistent pattern everywhere

Reply:
++ on removing duplicate code